### ISSUE TITLE
MotionSensors, lux, bug?

### ISSUE THREAD

**Issue by pointlevel** [Pythm/ad-Lightwand#4]
Hi,

I have found a strange behavior with the motionlights., When i have motionlights activated and the sensor for the OutLux updates, the lights turn off, no matter the value(i have tried 0.0-150.0). The mode is normal and i have this lux_constraint  for the light.

```yaml
      lux_constraint: 5000
      lux_turn_on: 4500
      lux_turn_off: 5000
```
The sensor updates every 5 minutes, and every time the motionlights that are activated turns off.

**Comment by Pythm** [comment]
I have not been able to replicate this error. The light is not suppose to turn on if the lux is above constraint. It could be some conversion error but you'll need to swap out some code to log the data from the lux sensor.

If you can swap out the out_lux_state function at line 570 with this one it creates logs to verify

```Python
    def out_lux_state(self, entity, attribute, old, new, kwargs) -> None:
        self.log(f"New lux: {new}")
        try:
            if self.outLux1 != float(new):
                self.outLux1 = float(new)
        except ValueError as ve:
            self.log(f"Not able to get new outlux. ValueError: {ve}", level = 'INFO')
        except TypeError as te:
            self.log(f"Not able to get new outlux. TypeError: {te}", level = 'INFO')
        except Exception as e:
            self.log(f"Not able to get new outlux. Exception: {e}", level = 'WARNING')
        else:
            self.log(f"Setting new lux: {self.outLux1}")
            self.newOutLux()
```

I can also check over the yaml configuration if you post it or send it to me.

**Comment by pointlevel** [comment]
Here is the logs from Appdaemon:

```
2024-11-03 16:22:54.469345 INFO Backporch: New lux: 17.4
2024-11-03 16:22:54.479554 INFO Diningroom: New lux: 17.4
2024-11-03 16:22:54.482353 INFO Hallway: New lux: 17.4
2024-11-03 16:22:54.483410 INFO Kitchen: New lux: 17.4
2024-11-03 16:22:54.502287 INFO MasterBedroom: New lux: 17.4
2024-11-03 16:22:54.503504 INFO Livingroom: New lux: 17.4
2024-11-03 16:22:54.505734 INFO Hallway: Setting new lux: 17.4
2024-11-03 16:22:54.508305 INFO Diningroom: Setting new lux: 17.4
2024-11-03 16:22:54.511526 INFO Backporch: Setting new lux: 17.4
2024-11-03 16:22:54.516041 INFO Kitchen: Setting new lux: 17.4
2024-11-03 16:22:54.517547 INFO MasterBedroom: Setting new lux: 17.4
2024-11-03 16:22:54.524055 INFO Livingroom: Setting new lux: 17.4
2024-11-03 16:22:54.576909 INFO Hallway: New lux: 12.6
2024-11-03 16:22:54.591069 INFO Diningroom: New lux: 12.6
2024-11-03 16:22:54.598024 INFO Hallway: Setting new lux: 12.6
2024-11-03 16:22:54.606102 INFO Backporch: New lux: 12.6
2024-11-03 16:22:54.607830 INFO Kitchen: New lux: 12.6
2024-11-03 16:22:54.609376 INFO MasterBedroom: New lux: 12.6
2024-11-03 16:22:54.610375 INFO Diningroom: Setting new lux: 12.6
2024-11-03 16:22:54.614507 INFO MasterBedroom: Setting new lux: 12.6
2024-11-03 16:22:54.620698 INFO Kitchen: Setting new lux: 12.6
2024-11-03 16:22:54.622853 INFO Backporch: Setting new lux: 12.6
2024-11-03 16:22:54.638283 INFO Livingroom: New lux: 12.6
2024-11-03 16:22:54.639051 INFO Livingroom: Setting new lux: 12.6
```

And here is my yaml for the hallway as an example, but every room acts the same way.

```yaml
Hallway:
  module: lightwand
  class: Room
  json_path: /homeassistant/.storage/

  OutLux_sensor: sensor.outdoor_lux

  motion_sensors:
    - motion_sensor: binary_sensor.activity_in_hallway
      delay: 10
    - motion_sensor: input_boolean.fake_activity_in_hallway
      delay: 10

  Lights:
    - lights:
        - light.hallway_motion_lights

      motionlights:
        - time: "00:00:00"
          light_data:
            brightness: 30
            transition: 3
        - time: "06:00:00"

      light_modes:
        - mode: on_hallway
          light_data:
            brightness: 255
            transition: 1
        - mode: off_hallway
          state: turn_off

      lux_constraint: 5000
      lux_turn_on: 4500
      lux_turn_off: 5000
```

**Comment by Pythm** [comment]
Thank you! With that configuration I found out what the bug was. I have only used the combination of motionlights and lux_turn_on / lux_turn_off in the entrance, but then in combination with automation and motion triggering an offset +35 so it was hidden from me. It is fixed in version 1.1.9 coming soon

Comments on configuration:
- lux_turn_on and lux_turn_off does not add anything unless you are using automations in addition to motionlights. 
- Instead off defining a mode "off_hallway" you can call off_"name" that is off_Hallway to turn off lights. Saves you some lines of configuration.
